staging/src/k8s.io/apimachinery/pkg/util/yaml/decoder.go

```diff
@@ -344,3 +379,14 @@ func hasPrefix(buf []byte, prefix []byte) bool {
 	trim := bytes.TrimLeftFunc(buf, unicode.IsSpace)
 	return bytes.HasPrefix(trim, prefix)
 }
+
+// sessionIsTTY returns true when the Stdin reader is reading from terminal
+// as opposed to having data being piped in
+func sessionIsTTY() bool {
```

Comment:
We already have helpers for this - it would be better to reuse those rather than hardcoding a new one here.  pkg/util/term.IsTerminal(writer)